config/config.go

```diff
@@ -136,3 +138,31 @@ func Parse(raw map[string]string) (Config, error) {
 
 	return config, nil
 }
+
+// GetOptions returns generated options for mongo connection depending on mechanism.
+func (d *Config) GetOptions() *options.ClientOptions {
+	uri, properties := d.getURIAndPropertiesByMechanism()
+	cred := options.Credential{
+		AuthMechanism:           string(d.Auth.Mechanism),
+		AuthMechanismProperties: properties,
+		AuthSource:              d.Auth.DB,
+		Username:                d.Auth.Username,
+		Password:                d.Auth.Password,
+		PasswordSet:             true,
```

Comment:
The `PasswordSet` is used only for the `GSSAPI` auth mechanism, but the Go mongo driver doesn't support this mechanism, and we can omit this field.